### PR TITLE
SIMSBIOHUB-559 BCTW Patch and Mortality Refactor

### PR DESCRIPTION
**Pull request by MacQSL**
### Changes
- Additional format for BCTW when querying multiple critters (getMultipleCrittersByIdsDetailed)
- Mortality service refactor, now repository + service classes
- Additional documentation for classes
- Updated swagger docs for mortality
- Test coverage

### CODE REVIEW

**Review by NickPhura — APPROVED** [review]
Looks good to me.

I ran it locally and tried adding/editing animals through SIMS (hopefully that hits most of these endpoints). Everything seemed to work as expected.

File: src/services/marking-service.ts [hunk 1]
```diff
@@ -1,8 +1,40 @@
...
+
+  /**
+   * Instantiate MarkingService and inject dependencies.
+   *
+   * @static
+   * @returns {MarkingService}
+   */
+  static init(): MarkingService {
+    return new MarkingService(new MarkingRepository(), new ItisService());
+  }
```
Comment (NickPhura) [inline src/services/marking-service.ts:36]: question: Is this just for convenience? So if you don't want to overwrite anything by calling the constructor, you can just call this?

Comment (NickPhura) [reply src/services/marking-service.ts:36]: I wonder what would happen if you called:

`new MarkingService(a, b).init()`?

Comment (NickPhura) [reply src/services/marking-service.ts:36]: Ah its static, nvm!

Comment (MacQSL) [reply src/services/marking-service.ts:36]: Yea just for convenience, anticipating some of the services (critter / bulk) to have quite a few dependencies. I kinda like it for the benefits of DI in regards to testing and the simplicity of instantiating (like if you just directly imported the dependencies)

**Pull request merged**